### DISCUSSION TITLE
Inquiry about additional points between each entry and exit points

### DISCUSSION THREAD

**Discussion by jackzhy96** [surgical-robotics-ai/surgical_robotics_challenge#25] [Q&A] [answered]
Hi Adnan,

I have a quick question regarding to challenge 3. According to the description, some additional points which is to define the desired circular path between each entry and exit point will be given. Nevertheless, I cannot find corresponding information in the repo. Would you mind letting me know where or how I can find it? 

In addition, may I ask how many points will be given for the circular trajectory? Thanks!

Best,
Haoying (Jack) Zhou

**Comment by adnanmunawar** [comment]
Hi Jack,
Discussing this with Dr. Kazanzides and will update you shortly.

**Comment by adnanmunawar** [comment, marked as answer]
Hi Jack. So I discussed this with @pkazanzides. You can access the pose (position + orientation) of each entry and exit point using the methods from [scene.py](https://github.com/collaborative-robotics/surgical_robotics_challenge/blob/master/scripts/surgical_robotics_challenge/scene.py) or the CRTK-ROS topics from [launch_crtk_interface.py](https://github.com/collaborative-robotics/surgical_robotics_challenge/blob/master/scripts/surgical_robotics_challenge/launch_crtk_interface.py). Once you have the pose, you can compute the midpoint for the optimal trajectory, such that the needle curvature remains parallel to the `XZ plane` of the entry/exit points and the needle rotates around the `Y axis` of the entry point. Please refer to the Entry/Exit frames section of the [scene_coordinate_frames](https://github.com/collaborative-robotics/surgical_robotics_challenge/blob/master/docs/scene_coordinate_frames.md) document for the coordinate frames of the entry/exit points.
